### PR TITLE
Calendar - add 'initialDate' prop

### PR DESCRIPTION
**Pull request by Inbal-Tish**
Calendar - add 'initialDate' prop that initializes the component when changed (will replace 'current' after migration)

### CODE REVIEW

File: src/calendar/index.tsx [hunk 1]
```diff
@@ -91,6 +93,8 @@ class Calendar extends Component<CalendarProps, CalendarState> {
     style: PropTypes.oneOfType([PropTypes.object, PropTypes.array, PropTypes.number]),
     /** Initially visible month. Default = Date() */
     current: PropTypes.any,
+    /** Initially visible month. If changed will initialize the calendar to this value */
+    initialDate: PropTypes.any,
```
Comment (ethanshar) [inline src/calendar/index.tsx:97]: I think we can kill 2 birds here. 
What if `initialDate` will be a timestamp (number) 
As we discussed before, as part of the big refactor, we'll want to use timestamp for our value source.

File: src/calendar/index.tsx [hunk 2]
```diff
@@ -132,12 +136,24 @@ class Calendar extends Component<CalendarProps, CalendarState> {
     enableSwipeMonths: false
   };
 
+  currentDate = this.props.current || this.props.initialDate;
+
   state = {
-    currentMonth: this.props.current ? parseDate(this.props.current) : new XDate()
+    currentMonth:
```
Comment (ethanshar) [inline src/calendar/index.tsx:142]: I know the code was already like that, but why are we setting the current date into something called `currentMonth`

File: src/calendar/index.tsx [hunk 3]
```diff
@@ -280,7 +296,7 @@ class Calendar extends Component<CalendarProps, CalendarState> {
 
   renderHeader() {
     const {customHeader, headerStyle, displayLoadingIndicator, markedDates, testID} = this.props;
-    const current = parseDate(this.props.current);
+    const current = parseDate(this.currentDate);
```
Comment (ethanshar) [inline src/calendar/index.tsx:299]: Why are you talking the current date from the `this` and not from the state? 
In general, why are we keeping the current date also on `this`?

File: src/calendar/index.tsx (hunk 2 above)
Comment (Inbal-Tish) [reply src/calendar/index.tsx:142]: The calendar only handles the month/year so that's is focus

File: src/calendar/index.tsx (hunk 3 above)
Comment (Inbal-Tish) [reply src/calendar/index.tsx]: Changed

File: src/calendar/index.tsx (hunk 1 above)
Comment (Inbal-Tish) [reply src/calendar/index.tsx:98]: Do you mean to change 'initialDate' type to a timestamp in preparation for the timestamp support? I will simply convert to XDate at this point...

Comment (Inbal-Tish) [reply src/calendar/index.tsx:98]: Did we decide if to support timestamp as number or string?

Comment (ethanshar) [reply src/calendar/index.tsx:98]: Keep the string we have today
and add a ticket to create a util for formatting the value string format

**Comment by Inbal-Tish** [comment]
@ethanshar I merged master and moved 'initialDate' to be string.

**Review by ethanshar — APPROVED** [review]

**Pull request merged**